src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemController.java

```diff
@@ -61,13 +63,13 @@ public interface CgroupSubsystemController {
     public static String getStringValue(CgroupSubsystemController controller, String param) {
         if (controller == null) return null;
 
-        try {
-            return CgroupUtil.readStringValue(controller, param);
-        }
-        catch (IOException e) {
+        Path filePath = Path.of(controller.path(), param);
+        try (BufferedReader bufferedReader = Files.newBufferedReader(filePath)) {
+            String line = bufferedReader.readLine();
+            return line;
+        } catch (IOException e) {
```

Comment:
Is the catch for `UncheckedIOException` due to some previously known failure resulting in the use of these APIs?

Reply:
Without using `Files.lines()` no `UncheckedIOException` would be thrown. Just `IOException` and `null` returned. This extra catch is there to avoid new `UncheckedIOException` being thrown on `firstLine()`. I.e. to keep semantics the same as before.

Reply:
Thank you that's a good point. I hadn't taken into account exceptions propagated from `java.util.stream.Stream` operations.